src/main/java/com/artipie/asto/streams/StorageValueAsIStream.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * The MIT License (MIT) Copyright (c) 2020-2021 artipie.com
+ * https://github.com/artipie/asto/LICENSE.txt
+ */
+package com.artipie.asto.streams;
+
+import com.artipie.asto.ArtipieIOException;
+import com.artipie.asto.Key;
+import com.artipie.asto.Storage;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.PipedInputStream;
+import java.io.PipedOutputStream;
+import java.util.concurrent.CompletionStage;
+import java.util.function.Function;
+import org.cqfn.rio.WriteGreed;
+import org.cqfn.rio.stream.ReactiveOutputStream;
+
+/**
+ * Process storage value as input stream.
+ * This class allows to treat storage item as input stream and
+ * perform some action with this stream (read/uncompress/parse etc).
+ * @param <T> Result type
+ * @since 1.4
+ */
+public final class StorageValueAsIStream<T> {
+
+    /**
+     * Abstract storage.
+     */
+    private final Storage asto;
+
+    /**
+     * Storage item key.
+     */
+    private final Key key;
+
+    /**
+     * Ctor.
+     * @param asto Abstract storage
+     * @param key Storage item key
+     */
+    public StorageValueAsIStream(final Storage asto, final Key key) {
```

Comment:
Storage abstraction looks unnecessary high here, we can provide only `Content` to convert it into input stream. This way we can also use this class not only for storage but for HTTP requests too.